python/src/tasks/differential_expression.py

```diff
@@ -62,26 +63,37 @@ def compute(self):
             raw_adata.obs["cell_ids"].isin(de_base)
         ] = "first"
 
-        # Do a pairwise wilcoxon test
-        result = de.test.pairwise(
-            data=raw_adata,
-            grouping="condition",
-            test="t-test",
-            lazy=False,
-            noise_model=None,
+        request = {
+            "baseCells": raw_adata.obs.index[
+                raw_adata.obs["condition"] == "first"
+            ].tolist(),
+            "backgroundCells": raw_adata.obs.index[
+                raw_adata.obs["condition"] == "second"
+            ].tolist(),
+        }
+
+        r = requests.post(
+            f"{config.R_WORKER_URL}/v0/DifferentialExpression",
+            headers={"content-type": "application/json"},
+            data=json.dumps(request),
         )
 
-        # massage into right format
-        result = result.summary_pairs(["first"], ["second"])
-        result = result[["gene", "pval", "qval", "log2fc"]]
-        result["gene_names"] = result["gene"]
-        del result["gene"]
+        with open("/data/result.json", "w") as f:
+            json.dump(r.json(), f)
+
+        result = pandas.DataFrame.from_dict(r.json())
+        result = result[["Z", "M", "Gene"]]
+        result = result.rename(
+            columns={"Z": "zscore", "M": "log2fc", "Gene": "gene_names"}
+        )
+
+        result["abszscore"] = result["zscore"].abs()
```

Comment:
could we not get `"Z"`, `"W"`, and `"Gene"` from R and then Python just sends them back to the API directly so that we don't do those transformations here?

Reply:
If we were to do the transformations from Python, then maybe it is worth deleting the `result.json` file or using `tempFile` to create it, since we don't need it after the `result` object is constructed.

Reply:
Sure, fixed.